backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -113,16 +114,19 @@ internal class ObjCExportCodeGenerator(
     fun FunctionGenerationContext.kotlinToObjC(
             value: LLVMValueRef,
             typeBridge: TypeBridge
-    ): LLVMValueRef = when (typeBridge) {
-        is ReferenceBridge -> kotlinReferenceToObjC(value)
-        is ValueTypeBridge -> kotlinToObjC(value, typeBridge.objCValueType)
-    }
+    ): LLVMValueRef = when {
+            LLVMTypeOf(value) == voidType -> kNullInt8Ptr
```

Comment:
This is incorrect, because Objective-C method can have non-pointer return type, e.g. if `Nothing`-returning Kotlin method overrides `Int`-returning method.